### ISSUE TITLE
Internal Server Error when trying to create a Tag in new SVN Repository

### ISSUE THREAD

**Issue by Shaoranlaos** [scm-manager/scm-manager#1665]
#### Describe how to reproduce the bug

1. Create a new initialized SVN Repository 
2. Checkout the trunk folder of the new repository
3. try to create a tag of the trunk in revision 1 under the path /tags/0.0.1 with "create-intermediate-folders"

#### Describe your SCM-Manager setup

SCM 2.18 with the following plugins:
scm-editor-plugin:2.2.1
scm-jenkins-plugin:2.2.1
scm-webhook-plugin:2.2.0
scm-hg-plugin:2.18.0
scm-issuetracker-plugin:3.1.0
scm-readme-plugin:2.0.0
scm-code-editor-plugin:1.0.0
scm-support-plugin:2.0.0
scm-branchwp-plugin:2.0.0
scm-git-plugin:2.18.0
scm-legacy-plugin:2.18.0
scm-markdown-plantuml-plugin:1.0.0
scm-jira-plugin:3.0.1
scm-pathwp-plugin:2.0.2
scm-el-plugin:1.0.0
scm-tagprotection-plugin:2.0.0
scm-archive-plugin:2.0.0
scm-landingpage-plugin:1.5.0
scm-mail-plugin:2.5.0
scm-repository-template-plugin:1.0.2
scm-svn-plugin:2.18.0
scm-authormapping-plugin:2.0.1
scm-ci-plugin:2.2.0
scm-ssh-plugin:2.3.0
scm-review-plugin:2.9.1
scm-ldap-plugin:2.3.0

The instance is proxied via a local nginx reverse proxy to make ssl configuration easier.

#### Which operating system, browser and versions do you use?

Server: CentOS 7, SCM 2.18
Client: TortoiseSVN 1.8.8 (also tried Eclipse with Subclipse newest Version as of writting this)

#### Log of the bug

Server: No Logmessages beside authoization messages

Client:
Error: COPY request on '/scm/repo/<namespace>/<repo>/!svn/bc/7/trunk' failed: 500 Server  

I also made a debug logging package (with the support plugin) but that contains to much private information to publish it here directly. If you need it please provide a non public way to get it to you.

#### Further information like screenshots
Tagging configuration dialog of TortoiseSVN
![Tortoise Tag Dialog](https://user-images.githubusercontent.com/25054970/119114538-8b170f80-ba26-11eb-93de-5d226a784ef0.png)

**Comment by sdorra** [comment]
Can you reproduce the problem with the command line client? At the moment i don't have access to a machine with TortoiseSVN, so i've tried to reproduce the problem with the command line client.

```bash
❯ svn copy -r 1 --parents https://scm-instance/scm/repo/issues/1665/trunk https://scm-instance/scm/repo/issues/1665/tags/1.0.0 -m 'Release 1.0.0'
Committing transaction...
Committed revision 2.
```

The command above creates a tag from revision 1 of the trunk folder, the `--parents` switch creates the intermediate directories.
The command runs without any error.

**Comment by sdorra** [comment]
The same with working copy:

```bash
❯ svn checkout https://scm-instance/scm/repo/issues/1665
A    1665/trunk
A    1665/trunk/README.md
Checked out revision 1.
❯ cd 1665
❯ svn copy -r 1 --parents trunk tags/1.0.0
A         tags
A         tags/1.0.0
A         tags/1.0.0/README.md
❯ svn commit -m 'Released 1.0.0'
Adding         tags
Adding         tags/1.0.0
Committing transaction...
Committed revision 2.
```

**Comment by Shaoranlaos** [comment]
I also tried via svn cli (1.7.16 newest version on centos 7) but got the same error.
svn: E175002: Server sent unexpected return value (500 Server Error) in response to COPY request for '/scm/repo/test/<repo>/!svn/bc/7/trunk'

But it seems to be something with our installation specifically. I tried it with another installation (a private one) and there the error does not happen (with the same clients).

Not sure what on our installation could be the problem. I also can not simply remove plugins until it works because the instance is already in productive use.

**Comment by markkrj** [comment]
If you followed [default documentation](https://www.scm-manager.org/docs/2.18.x/en/administration/reverse-proxies/), try removing ":$server_port" from your nginx config.

**Comment by Shaoranlaos** [comment]
> 
> 
> If you followed [default documentation](https://www.scm-manager.org/docs/2.18.x/en/administration/reverse-proxies/), try removing ":$server_port" from your nginx config.

That helped immediatly.
Will that cause any other problems?

**Comment by markkrj** [comment]
I don't think so. @sdorra know better, but I think SCM-Manager construct an "Host" header based on X-Forwarded-Host and pass it to SVNKIT. As I understand it, the "Host" header must match what is configured on SCM-Manager settings, but as I had configured it like "https://scm.example.com" and the "Host" header was being passed to SVNKIT as "https://scm.example.com:443", it was being denied. I did some digging to discover it, but can't remember exactly.

IMO, the ":$server_port" should be removed from documentation, or an explanation should be made that it must only be set if using non-regular ports and that it must match the scm-manager configuration.

**Comment by sdorra** [comment]
This is strange, the url should work with and without the port. I will do some tests.
